### ISSUE TITLE
[GCC 8] error: catching polymorphic type by value

### ISSUE THREAD

**Issue by cuviper** [WebAssembly/binaryen#1399]
I ran into this problem compiling Rust, but reproduced it on binaryen master (c81857a3a7708738b20bb28a320fc971e74626a7).

Compiling with GCC 8 on Fedora rawhide gives the following errors:

```
[  5%] Building CXX object src/wasm/CMakeFiles/wasm.dir/wasm-s-parser.cpp.o
/builddir/binaryen/src/wasm/wasm-s-parser.cpp: In member function ‘wasm::Name wasm::SExpressionWasmBuilder::getLabel(wasm::Element&)’:
/builddir/binaryen/src/wasm/wasm-s-parser.cpp:1411:19: error: catching polymorphic type ‘class std::invalid_argument’ by value [-Werror=catch-value=]
     } catch (std::invalid_argument) {
                   ^~~~~~~~~~~~~~~~
/builddir/binaryen/src/wasm/wasm-s-parser.cpp:1413:19: error: catching polymorphic type ‘class std::out_of_range’ by value [-Werror=catch-value=]
     } catch (std::out_of_range) {
                   ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
gmake[3]: *** [src/wasm/CMakeFiles/wasm.dir/build.make:159: src/wasm/CMakeFiles/wasm.dir/wasm-s-parser.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:1012: src/wasm/CMakeFiles/wasm.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:215: CMakeFiles/binaryen.dir/rule] Error 2
gmake: *** [Makefile:203: binaryen] Error 2
```

However, compiling without `-Werror` is successful. (cc #1387)

**Comment by kripken** [comment]
Thanks for reporting @cuviper , #1401 should fix it.

**Comment by cuviper** [comment]
I also fixed it -- #1400. There were a couple more locations that came up after fixing the first two errors.

**Comment by kripken** [comment]
Oh thanks, sorry I didn't see yours before.
